sequencers/modeshape-sequencer-epub/src/main/java/org/modeshape/sequencer/epub/EpubMetadata.java

```diff
@@ -0,0 +1,346 @@
+     */
+    private List<String> getRootfiles( ZipInputStream zipStream ) throws Exception {
+        List<String> rootfiles = new ArrayList<String>();
+        ZipEntry entry = null;
+        while ((entry = zipStream.getNextEntry()) != null) {
+            String entryName = entry.getName();
+            if (entryName.endsWith("META-INF/container.xml")) {
+                ByteArrayOutputStream content = getZipEntryContent(zipStream, entry);
+
+                DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+                DocumentBuilder builder = factory.newDocumentBuilder();
+                Document doc = builder.parse(new ByteArrayInputStream(content.toByteArray()));
+
+                XPathFactory xPathfactory = XPathFactory.newInstance();
+                XPath xpath = xPathfactory.newXPath();
+                XPathExpression expr = xpath.compile("/container/rootfiles/rootfile");
+                NodeList rootfileNodes = (NodeList) expr.evaluate(doc, XPathConstants.NODESET);
+
+                for (int i = 0; i < rootfileNodes.getLength(); i++) {
+                    Node node = rootfileNodes.item(i);
+                    rootfiles.add(node.getAttributes().getNamedItem("full-path").getNodeValue());
+                }
+                break;
+            }
+        }
+        return rootfiles;
+    }
+
+    /**
+     * Read the content of the ZipEntry without closing the stream.
+     */
+    private ByteArrayOutputStream getZipEntryContent(
+            ZipInputStream zipStream,
+            ZipEntry entry ) throws IOException {
+        ByteArrayOutputStream content = new ByteArrayOutputStream();
+        byte[] bytes = new byte[(int) entry.getSize()];
+        int read;
+        while ((read = zipStream.read(bytes, 0, bytes.length)) != -1) {
+            content.write(bytes, 0, read);
+        }
+        content.flush();
+        return content;
+    }
+
+    public List<EpubMetadataProperty> getTitle() {
+        return title;
+    }
+
+    public List<EpubMetadataProperty> getCreator() {
+        return creator;
+    }
+
+    public List<EpubMetadataProperty> getContributor() {
+        return contributor;
+    }
+
+    public List<EpubMetadataProperty> getLanguage() {
+        return language;
+    }
+
+    public List<EpubMetadataProperty> getIdentifier() {
+        return identifier;
+    }
+
+    public List<EpubMetadataProperty> getDescription() {
+        return description;
+    }
+
+    public List<EpubMetadataProperty> getPublisher() {
+        return publisher;
+    }
+
+    public List<EpubMetadataProperty> getRights() {
+        return rights;
+    }
+
+    public List<EpubMetadataProperty> getDate() {
+        return date;
+    }
+
+    public List<EpubMetadataProperty> getAlternateScript() {
+        return alternateScript;
+    }
+
+    /**
+     * Simple NamespaceContext which can resolve single namespace.
+     */
+    private class SimpleNSC implements NamespaceContext {
```

Comment:
there's already a `org.modeshape.common.xml.SimpleNamespaceContext` in `modeshape-common`. Perhaps we can use that here instead of having this inner class.